### PR TITLE
Transfer multiple inputs rebase

### PR DESCRIPTION
**Pull request by TimDaub**
Rebase of #120

Don't know if working appropriately.

### CODE REVIEW

**Comment by manolodewiner** [comment]
As there are breaking changes, we will need to update everything. So I propose to handle this PR next week after we first create a new release of the JS driver including the metadata search: #124 for this week. So:
- This week new release of JS driver including and five-bells-conditions repository #124
- Next week new realese of JS driver including this PR #125 . Update of any tutorial is needed

File: docs/Makefile [hunk 1]
```diff
@@ -26,7 +26,19 @@ html:
 	@echo
 	@echo "Build finished. The HTML pages are in $(BUILDDIR)/html."
 
+<<<<<<< dfe221031eab9e09c54fb4c8bf0feb54fd250043
```
Comment (TimDaub) [inline docs/Makefile]: rm merge conflict

**Comment by manolodewiner** [comment]
When creating a transfer transaction the user has to specify the 'transaction' and the 'output_index' that he/she is spending. For example:
```js
Transaction.makeTransferTransaction(
        [{ tx: transferTx, output_index: 0 }],
        [aliceOutput],
        metaData
    )
```
Travis do not mark this as an error, but maybe we want to change to camelcase:
```js
Transaction.makeTransferTransaction(
        [{ tx: transferTx, outputIndex: 0 }],
        [aliceOutput],
        metaData
    )
```

**Comment by manolodewiner** [comment]
Anyways I would like to merge this PR. Is needed for sub-divisible assets. Can you @jernejpregelj check it?

**Pull request merged**